packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -198,145 +98,262 @@ export function Comments( {
+							clearThreadFocus={ clearThreadFocus }
+						/>
+					</VStack>
+				) ) }
+		</>
+	);
+}
+
+function Thread( {
+	thread,
+	onEditComment,
+	onAddReply,
+	onCommentDelete,
+	onCommentResolve,
+	isFocused,
+	clearThreadFocus,
+} ) {
+	return (
+		<>
+			<CommentBoard
+				thread={ thread }
+				onResolve={ onCommentResolve }
+				onEdit={ onEditComment }
+				onDelete={ onCommentDelete }
+				status={ thread.status }
+			/>
+			{ 0 < thread?.reply?.length && (
+				<>
+					{ ! isFocused && (
+						<VStack className="editor-collab-sidebar-panel__show-more-reply">
+							{ sprintf(
+								// translators: 1: number of replies.
+								_x(
+									'%s more replies..',
+									'Show replies button'
+								),
+								thread?.reply?.length
+							) }
+						</VStack>
+					) }
+
+					{ isFocused &&
+						thread.reply.map( ( reply ) => (
+							<VStack
+								key={ reply.id }
+								className="editor-collab-sidebar-panel__child-thread"
+								id={ reply.id }
+								spacing="2"
+							>
+								{ 'approved' !== thread.status && (
 									<CommentBoard
 										thread={ reply }
-										parentThread={ thread }
+										onEdit={ onEditComment }
+										onDelete={ onCommentDelete }
 									/>
-								</VStack>
-							) ) }
-						{ 'reply' === actionState?.action &&
-							thread.id === actionState?.id && (
-								<VStack
-									className="editor-collab-sidebar-panel__child-thread"
-									spacing="2"
-								>
-									<HStack
-										alignment="left"
-										spacing="3"
-										justify="flex-start"
-									>
-										<CommentAuthorInfo />
-									</HStack>
-									<VStack
-										spacing="3"
-										className="editor-collab-sidebar-panel__comment-field"
-									>
-										<CommentForm
-											onSubmit={ ( inputComment ) => {
-												onAddReply(
-													inputComment,
-													thread.id
-												);
-												setActionState( false );
-											} }
-											onCancel={ () =>
-												setActionState( false )
-											}
-											submitButtonText={ _x(
-												'Reply',
-												'Add reply comment'
-											) }
-										/>
-									</VStack>
-								</VStack>
+								) }
+								{ 'approved' === thread.status && (
+									<CommentBoard thread={ reply } />
+								) }
+							</VStack>
+						) ) }
+				</>
+			) }
+			{ 'approved' !== thread.status && isFocused && (
+				<VStack
+					className="editor-collab-sidebar-panel__child-thread"
+					spacing="2"
+				>
+					<HStack alignment="left" spacing="3" justify="flex-start">
+						<CommentAuthorInfo />
+					</HStack>
+					<VStack
+						spacing="3"
+						className="editor-collab-sidebar-panel__comment-field"
+					>
+						<CommentForm
+							onSubmit={ ( inputComment ) => {
+								onAddReply( inputComment, thread.id );
+							} }
+							onCancel={ ( event ) => {
+								event.stopPropagation(); // Prevent the parent onClick from being triggered
```

Comment:
If we really need to do this, there must be really good reason for it. Why is there a click handler in a parent element. Is it focusable? Why not base comment focus on whether or not the wrapper or an inner focusable element is focussed?

Reply:
We are removing the board's focus when the cancel button is clicked which conflicts with the parent element click event. I added a focus event to the wrapper element and the focus state in the parent component because we require one board focused at a time.